ddl_checker/main.go

```diff
@@ -0,0 +1,216 @@
+		"and delete the conflict table\n" +
+		"Query mode: The program will ask you before synchronizing the dependent table structure from MYSQL\n" +
+		"Manual mode: This program does not perform anything other than executing the input SQL.\n" +
+		"SETMOD Usage: SETMOD <MODCODE>; MODCODE = [0(Auto), 1(Query), 2(Manual)]\n"
+)
+
+func main() {
+	initialise()
+	printWelcome()
+	mainLoop()
+	//handler("ALTER TABLE table_name MODIFY column_1 int(11) NOT NULL;")
+	destroy()
+}
+
+func initialise() {
+	var err error
+	reader = bufio.NewReader(os.Stdin)
+	executableChecker, err = ddl_checker.NewExecutableChecker()
+	if err != nil {
+		fmt.Printf("[DDLChecker]Init failed,Can't create ExecutableChecker: %s\n", err.Error())
+		os.Exit(1)
+	}
+	executableChecker.Execute("use test;")
+	db, err = sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%d)/%s",
+		*username, *password, *host, *port, *schema))
+	if err != nil {
+		fmt.Printf("[DDLChecker]Init failed,Can't open mysql database: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+}
+
+func destroy() {
+	executableChecker.Close()
+	db.Close()
+}
+
+func mainLoop() {
+	var input string
+	var err error
+	for cont := true; cont; cont = handler(input) {
+		fmt.Printf("[%s] > ", modeName())
+		input, err = reader.ReadString(';')
+		if err != nil {
+			fmt.Printf("[DDLChecker]Read Stdin Error: %s\n", err.Error())
+			os.Exit(1)
+		}
+	}
+}
+func handler(input string) bool {
+	// cmd exit
+	lowerTrimInput := strings.ToLower(strings.TrimSpace(input[:len(input)-1]))
+	if lowerTrimInput == "exit" {
+		return false
+	}
+	// cmd setmod
+	if strings.HasPrefix(lowerTrimInput, "setmod") {
+		modCodeTmp, err := strconv.Atoi(strings.TrimSpace(lowerTrimInput[6:]))
+		if err != nil || modCodeTmp > 2 || modCodeTmp < 0 {
+			fmt.Printf("[DDLChecker]SETMOD Usage: SETMOD <MODCODE>; MODCODE = [0(Auto), 1(Query), 2(Manual)]\n")
+		} else {
+			modCode = modCodeTmp
+		}
+		return true
+	}
+	if modCode != 2 {
+		// auto and query mod
+		stmt, err := executableChecker.Parse(input)
+		if err != nil {
+			fmt.Printf("[DDLChecker]SQL Parse Error: %s\n", err.Error())
+			return true
+		}
+		neededTable := ddl_checker.GetTableNeededExist(stmt)
+		nonNeededTable := ddl_checker.GetTableNeededNonExist(stmt)
+		// query mod
+		if modCode == 1 && !queryAutoSync(neededTable, nonNeededTable) {
+			goto EXECUTE
+		}
+		syncTablesFromMysql(neededTable)
+		dropTables(nonNeededTable)
+	}
+EXECUTE:
+	err := executableChecker.Execute(input)
+	if err == nil {
+		fmt.Println("[DDLChecker] SQL execution succeeded")
+	} else {
+		fmt.Println("[DDLChecker] SQL execution failed:", err.Error())
+	}
+	return true
+}
+
+func syncTablesFromMysql(tableNames []string) {
+	for _, tableName := range tableNames {
+		fmt.Println("[DDLChecker] Syncing Table", tableName)
+		isExist := executableChecker.IsTableExist(tableName)
+		if isExist {
+			fmt.Println("[DDLChecker] Table", tableName, "is exist,Skip")
+			continue
+		}
+		rows, err := db.Query("show create table " + tableName)
```

Comment:
1. you could use `QueryRow` here
2. you need to escape the table name, at the minimum:

    ```go
    "SHOW CREATE TABLE `" + tableName + "`"
    ```

    there might  be an existing function here to add the quotation marks.